### ISSUE TITLE
Cannot install with cocoapods without use_frameworks

### ISSUE THREAD

**Issue by amloelxer** [DataDog/dd-sdk-ios#131]
### The issue

Hey all, first thanks so much working on this, I'm excited to give it a whirl in our app. When trying to install with cocoapods, I get the following error: `Using Swift static libraries with custom module maps is currently not supported. Please build DatadogSDK as a framework or remove the custom module map`. Unfortunately for my team, we can't move to `use_frameworks` and need to keep modules. When looking at your custom Datadog module map, I see it is eerily similar to an autogenerated one with the exception of the sudo private module you have set exposing `_Datadog_Private`
```
framework module _Datadog_Private {
  export *
  header "ObjcExceptionHandler.h"
}
```
Any reason this can't just be exposed in DatadogObjc and let the autogenerated module map be used (other than the fact you want to discourage 3d party devs from consuming `ObjcExceptionHandler.m`)? Or any easy workarounds short of manually integrating into our project? Thank you for your time. 

- Alex 

---

#### Datadog SDK version:

DatadogSDK (1.2.1)

#### Last working Datadog SDK version:

First time using

#### Dependency Manager:

Cocoapods 1.9.1

#### Other toolset:

N / A

#### Xcode version:

Version 11.5 (11E608c)

#### Swift version:

Swift 5 

#### Deployment Target:

iOS 11

#### macOS version:

Mac OS Catalina v 10.15.2

**Comment by buranmert** [comment]
hi @amloelxer 👋  thanks for reporting and looking into the issue!

you are right, [we use a custom `modulemap` in order to use ObjC in Swift while hiding it from our users](https://github.com/DataDog/dd-sdk-ios/pull/80). 
unfortunately, [we can't remove it at the moment](https://github.com/DataDog/dd-sdk-ios/pull/64).

you can use Carthage or Swift Package Manager without this problem, but i assume you use only Cocoapods in your projects

you can try [the snippet below in your `Podfile`s](https://stackoverflow.com/a/60914505/1515075):
```ruby
target 'Your-Target-Name' do
  use_frameworks!
  pod 'DatadogSDK'
  pod 'YourOtherPod'
  ...
  target 'Your-TestTarget-Name' do
    inherit! :search_paths
    pod 'YourTestPod'
    ...
  end
end

dynamic_frameworks = ['DatadogSDK','DatadogSDKObjc']
pre_install do |installer|
  installer.pod_targets.each do |pod|
    if !dynamic_frameworks.include?(pod.name)
      def pod.static_framework?;
        true
      end
      def pod.build_type;
        Pod::BuildType.static_library
      end
    end
  end
end
```
this code should keep **all your other pods static** while _DatadogSDK being dynamic_

✅ i verified the fix by using `Alamofire` statically and `Datadog` dynamically in the same project with this code
<img width="269" alt="Screenshot 2020-06-11 at 14 04 08" src="https://user-images.githubusercontent.com/1417500/84383394-7ca45f00-abec-11ea-84ed-4ecbec953791.png">


i hope that solves your problems @amloelxer and @hyling 🤞 
please let us know how that plays out

**Comment by amloelxer** [comment]
@buranmert thanks so much for the response! I've been swamped with work but will look into this as soon as I have a free moment.

**Comment by buranmert** [comment]
hi @amloelxer and @hyling, have you had a chance to see if that works for you?

**Comment by hyling** [comment]
@buranmert Carthage along side Cocoapods works for me!  Thanks for the solution.

**Comment by buranmert** [comment]
@hyling glad to hear that!
@amloelxer i'm closing the issue for now, but feel free to reopen if needed

**Comment by buranmert** [comment]
ℹ️  this issue is fixed in `DatadogSDK v1.5.2`
